openedx/core/djangoapps/schedules/README.rst

```diff
@@ -34,29 +34,36 @@ The app introduces a Schedule object to the edX codebase. Learners receive
 Schedules when they enroll in self-paced courses. The app uses Schedules to
 determine which learners to message.
 
+In the future, the Schedules app may be extended or changed to support more
+complicated communication patterns. For example, something that takes into
+account the user's progress through the course.
+
 Glossary
 --------
 
--  **Schedule**: Stores the day a learner enrolls in a course and the learner’s
-“upgrade deadline”. Used by the app at runtime to determine which learners to
-message.
+-  **Schedule**: Stores the day a learner enrolls in a course and the
+   learner's "upgrade deadline". This information allows us to personalize
+   learners' experiences in self-paced courses so that events happen relative to
+   the learner's schedule, not the course's schedule.
 
 -  **Schedule Experience**: Defines the set of emails that a learner
    receives. The two Schedule Experiences are:
 
    -  “Recurring Nudge and Upgrade Reminder”
    -  “Course Updates”
 
--  **Upgrade Deadline**: The date before which a learner can purchase a
-   verified certificate. By default, a Schedule imposes a “soft” upgrade
-   deadline 21 days from when a learner enrolled in a course. A course
-   also imposes a “hard” upgrade deadline. A Schedule uses whichever
-   date is earlier.
+-  **Upgrade Deadline**: The date before which a learner is encouraged to
+   purchase a verified certificate. By default, a Schedule imposes a "soft"
+   upgrade deadline (meaning, a suggested, but not final, date) 21 days from
```

Comment:
I would suggest that we state "Learners are able to upgrade after the soft deadline. They cannot upgrade after the course-wide expiration.".

Reply:
We should figure out what we're allowed to message here. We shouldn't state the fact that this deadline is soft vs. hard if we're trying to conceal this fact.

How open can we be here, @mulby?

Reply:
We can be open!